File: app/src/main/java/com/vikingsen/cheesedemo/model/prefs/Prefs.java
package com.vikingsen.cheesedemo.model.prefs;


import android.content.SharedPreferences;

import javax.inject.Inject;
import javax.inject.Singleton;

import okhttp3.logging.HttpLoggingInterceptor;

@Singleton
public class Prefs {

    private static final String PREF_HTTP_LOG_LEVEL = "dev_http_log_level";

    private final SharedPreferences preferences;

    @Inject
    Prefs(SharedPreferences preferences) {
        this.preferences = preferences;
    }

    public HttpLoggingInterceptor.Level getHttpLoggingLevel() {
        String logLevelAsString = preferences.getString(PREF_HTTP_LOG_LEVEL, HttpLoggingInterceptor.Level.BASIC.name());
        return HttpLoggingInterceptor.Level.valueOf(logLevelAsString);
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/model/prefs/Prefs.kt
package com.vikingsen.cheesedemo.model.prefs


import android.content.SharedPreferences
import okhttp3.logging.HttpLoggingInterceptor
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Prefs
@Inject constructor(private val preferences: SharedPreferences) {

    val httpLoggingLevel: HttpLoggingInterceptor.Level
        get() {
            val logLevelAsString = preferences.getString(PREF_HTTP_LOG_LEVEL, null) ?: return HttpLoggingInterceptor.Level.BASIC
            return HttpLoggingInterceptor.Level.valueOf(logLevelAsString)
        }

    companion object {

        private const val PREF_HTTP_LOG_LEVEL = "dev_http_log_level"
    }
}
